rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionInstant.java

```diff
@@ -0,0 +1,121 @@
+ */
+
+package co.rsk.rpc.modules.eth;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.core.Wallet;
+import co.rsk.mine.MinerClient;
+import co.rsk.mine.MinerServer;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.core.*;
+import org.ethereum.rpc.TypeConverter;
+import org.ethereum.rpc.Web3;
+import org.ethereum.rpc.exception.JsonRpcInvalidParamException;
+import org.ethereum.vm.GasCost;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+
+import static org.ethereum.rpc.TypeConverter.stringHexToByteArray;
+
+public class EthModuleTransactionInstant implements EthModuleTransaction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("web3");
+
+    private final RskSystemProperties config;
+    private final Wallet wallet;
+    private final TransactionPool transactionPool;
+    private final MinerServer minerServer;
+    private final MinerClient minerClient;
+    private final Blockchain blockchain;
+
+    public EthModuleTransactionInstant(RskSystemProperties config, Wallet wallet, TransactionPool transactionPool, MinerServer minerServer, MinerClient minerClient, Blockchain blockchain) {
+        this.config = config;
+        this.wallet = wallet;
+        this.transactionPool = transactionPool;
+        this.minerServer = minerServer;
+        this.minerClient = minerClient;
+        this.blockchain = blockchain;
+    }
+
+    @Override
+    public synchronized String sendTransaction(Web3.CallArguments args) {
+        Account account = this.wallet.getAccount(new RskAddress(args.from));
+        String s = null;
+        try {
+            String toAddress = args.to != null ? Hex.toHexString(stringHexToByteArray(args.to)) : null;
+
+            BigInteger value = args.value != null ? TypeConverter.stringNumberAsBigInt(args.value) : BigInteger.ZERO;
+            BigInteger gasPrice = args.gasPrice != null ? TypeConverter.stringNumberAsBigInt(args.gasPrice) : BigInteger.ZERO;
+            BigInteger gasLimit = args.gas != null ? TypeConverter.stringNumberAsBigInt(args.gas) : BigInteger.valueOf(GasCost.TRANSACTION_DEFAULT);
+
+            if (args.data != null && args.data.startsWith("0x")) {
+                args.data = args.data.substring(2);
+            }
+
+            Transaction tx;
+
+            synchronized (transactionPool) {
+                BigInteger accountNonce = args.nonce != null ? TypeConverter.stringNumberAsBigInt(args.nonce) : transactionPool.getPendingState().getNonce(account.getAddress());
+                tx = Transaction.create(config, toAddress, value, accountNonce, gasPrice, gasLimit, args.data);
+                tx.sign(account.getEcKey().getPrivKeyBytes());
+                transactionPool.addTransaction(tx.toImmutableTransaction());
+                s = tx.getHash().toJsonString();
+            }
+
+            mineTransaction();
```

Comment:
this are the same validations and actions with the transaction pool as in `EthModuleTransactionBase` i would recomend moving that to another method and  sendTransaction should call that method, that way we won't repeat  code

Reply:
I know we are duplicating code, but is not to much code.
If we do the change you are requesting, it implies create a new class/methods. To extracts validations we need to return multiples outputs. What we can do is a basic class inheireted by the classes you said but i'm not super fan of that. If you have a better idea let me know.